### PR TITLE
Version 0.12.4: Add Rails 5 support; Drop Rails 4.1 specs

### PR DESCRIPTION
**Pull request by foobear**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,6 +4,7 @@ rvm:
   - "1.8.7"
   - "2.1.0"
   - "2.2.6"
```
Comment (triskweline) [inline .travis.yml]: Let's kill Ruby 2.2.

File: README.md [hunk 2]
```diff
@@ -8,7 +8,7 @@ Consul is an authorization solution for Ruby on Rails where you describe *sets o
 We have used Consul in combination with [assignable_values](https://github.com/makandra/assignable_values) to solve a variety of authorization requirements ranging from boring to bizarre.
 Also see our crash course video: [Solving bizare authorization requirements with Rails](http://bizarre-authorization.talks.makandra.com/).
 
-Consul is tested with Rails 2.3, 3.2, 4.1, and 4.2.7 on Ruby 1.8.7, 2.1.0, and 2.2.6.
+Consul is tested with Rails 2.3, 3.2, 4.1, 4.2.7, and 5.1.1 on Ruby 1.8.7, 2.1.0, 2.2.6, and 2.4.1 (only if supported, for each Ruby/Rails combination).
```
Comment (triskweline) [inline README.md]: Remove 4.1 and all the patch versions.

File: spec/rails-5.1.1/.gitignore [hunk 3]
```diff
@@ -0,0 +1 @@
+/log/*.log
```
Comment (triskweline) [inline spec/rails-5.1.1/.gitignore]: Can we name the folder `rails-5.1` without the patch level, like the others.

File: spec/rails-2.3/spec/spec_helper.rb [hunk 4]
```diff
@@ -26,11 +18,19 @@
 Dir[File.expand_path(File.join(File.dirname(__FILE__),'support','**','*.rb'))].each {|f| require f}
 
 # Run the migrations
+ConsulMigration = ActiveRecord::Migration
 print "\033[30m" # dark gray text
 ActiveRecord::Migrator.migrate("#{Rails.root}/db/migrate")
 print "\033[0m"
 
+module ControllerSpecHelpers
+  def wrap_params(params)
```
Comment (triskweline) [inline spec/rails-2.3/spec/spec_helper.rb]: Can we add a comment for future devs explaining why we need this in controller specs.

File: spec/shared/app_root/app/controllers/songs_controller.rb [hunk 5]
```diff
@@ -8,6 +8,7 @@ def show
   end
 
   def update
+    byebug
```
Comment (triskweline) [inline spec/shared/app_root/app/controllers/songs_controller.rb]: Forgot this line? Why doesn't this break Travis?

File: lib/consul/controller.rb [hunk 6]
```diff
@@ -43,8 +43,10 @@ def require_power_check(options = {})
       def skip_power_check(options = {})
         if Rails.version.to_i < 4
           skip_before_filter :unchecked_power, options
-        else
+        elsif Rails.version.to_i < 5
           skip_before_action :unchecked_power, options
+        else
+          skip_before_action :unchecked_power, { :raise => false }.merge!(options)
```
Comment (triskweline) [inline lib/consul/controller.rb]: What does `:raise => false` do? I couldn't find info about it in the code or Rails docs. Maybe explain in a comment.

**Comment by foobear** [comment]
This PR was closed since it was opened only so Travis would build.

I'll adjust as a new commit.

File: README.md (hunk 2 above)
Comment (foobear) [inline README.md]: Rails 4.1 spec app was removed already. I'll update the docs.

File: spec/rails-5.1.1/.gitignore (hunk 3 above)
Comment (foobear) [inline spec/rails-5.1.1/.gitignore]: `rails-5.1.1` directory was aligned with naming of `rails-4.2.7` directory. I'll also rename that to `rails-4.2` then.

File: lib/consul/controller.rb (hunk 6 above)
Comment (foobear) [inline lib/consul/controller.rb]: https://github.com/rails/rails/blob/master/activesupport/lib/active_support/callbacks.rb#L701.L709

I'll add a comment.

File: spec/shared/app_root/app/controllers/songs_controller.rb (hunk 5 above)
Comment (foobear) [inline spec/shared/app_root/app/controllers/songs_controller.rb]: Yes, it's an oversight. I don't know why Travis would not complain.

File: lib/consul/controller.rb (hunk 6 above)
Comment (foobear) [inline lib/consul/controller.rb]: FYI, it's documented in the Rails docs for `skip_callback` (which is called by `skip_before_action`): http://api.rubyonrails.org/classes/ActiveSupport/Callbacks/ClassMethods.html#method-i-skip_callback

**Pull request closed without merging**